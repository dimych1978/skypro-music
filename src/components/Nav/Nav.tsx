'use client';

import Image from 'next/image';
import styles from './Nav.module.css';
import React from 'react';
import Link from 'next/link';

export const Logo = () => {
  return (
    <div className={styles.navLogo}>
      <Image
        className={styles.logoImage}
        src='/img/logo.png'
        alt='logo'
        width={250}
        height={170}
      />
    </div>
  );
};

export const Nav = () => {
  const [display, setDisplay] = React.useState<string>('none');

  const hideDisplay: React.MouseEventHandler<HTMLDivElement> = () => {
    setDisplay(display === 'inline-block' ? 'none' : 'inline-block');
  };

  return (
    <nav className={styles.mainNav}>
      <Logo />
      <div className={styles.navBurger} onClick={hideDisplay}>
        <span className={styles.burgerLine}></span>
        <span className={styles.burgerLine}></span>
        <span className={styles.burgerLine}></span>
      </div>
      <div className={styles.navMenu} style={{ display: display }}>
        <ul className={styles.menuList}>
          <li className={styles.menuItem}>
            <Link href='/trackPages' className={styles.menuLink}>
              Главное
            </Link>
          </li>
          <li className={styles.menuItem}>
            <Link href='/trackPages/favorite' className={styles.menuLink}>
              Мой плейлист
            </Link>
          </li>
          <li className={styles.menuItem}>
            <Link href='/entryPages/login' className='menu__link'>
              Войти
            </Link>
          </li>
        </ul>
      </div>
    </nav>
  );
};
