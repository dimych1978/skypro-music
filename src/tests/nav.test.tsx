import { render } from '@testing-library/react';
import { Nav } from '@/components/Nav/Nav';

it('renders correctly', () => {
  const tree = render(<Nav />);
  expect(tree).toMatchSnapshot();
});
