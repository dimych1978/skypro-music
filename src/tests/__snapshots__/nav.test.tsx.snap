// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="mainNav"
      >
        <div
          class="navLogo"
        >
          <img
            alt="logo"
            class="logoImage"
            data-nimg="1"
            decoding="async"
            height="170"
            loading="lazy"
            src="/_next/image?url=%2Fimg%2Flogo.png&w=640&q=75"
            srcset="/_next/image?url=%2Fimg%2Flogo.png&w=256&q=75 1x, /_next/image?url=%2Fimg%2Flogo.png&w=640&q=75 2x"
            style="color: transparent;"
            width="250"
          />
        </div>
        <div
          class="navBurger"
        >
          <span
            class="burgerLine"
          />
          <span
            class="burgerLine"
          />
          <span
            class="burgerLine"
          />
        </div>
        <div
          class="navMenu"
          style="display: none;"
        >
          <ul
            class="menuList"
          >
            <li
              class="menuItem"
            >
              <a
                class="menuLink"
                href="/trackPages"
              >
                Главное
              </a>
            </li>
            <li
              class="menuItem"
            >
              <a
                class="menuLink"
                href="/trackPages/favorite"
              >
                Мой плейлист
              </a>
            </li>
            <li
              class="menuItem"
            >
              <a
                class="menu__link"
                href="/entryPages/login"
              >
                Войти
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="mainNav"
    >
      <div
        class="navLogo"
      >
        <img
          alt="logo"
          class="logoImage"
          data-nimg="1"
          decoding="async"
          height="170"
          loading="lazy"
          src="/_next/image?url=%2Fimg%2Flogo.png&w=640&q=75"
          srcset="/_next/image?url=%2Fimg%2Flogo.png&w=256&q=75 1x, /_next/image?url=%2Fimg%2Flogo.png&w=640&q=75 2x"
          style="color: transparent;"
          width="250"
        />
      </div>
      <div
        class="navBurger"
      >
        <span
          class="burgerLine"
        />
        <span
          class="burgerLine"
        />
        <span
          class="burgerLine"
        />
      </div>
      <div
        class="navMenu"
        style="display: none;"
      >
        <ul
          class="menuList"
        >
          <li
            class="menuItem"
          >
            <a
              class="menuLink"
              href="/trackPages"
            >
              Главное
            </a>
          </li>
          <li
            class="menuItem"
          >
            <a
              class="menuLink"
              href="/trackPages/favorite"
            >
              Мой плейлист
            </a>
          </li>
          <li
            class="menuItem"
          >
            <a
              class="menu__link"
              href="/entryPages/login"
            >
              Войти
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
